core/src/main/java/com/microsoft/applicationinsights/internal/logger/InternalLogger.java

```diff
@@ -197,6 +207,21 @@ public void trace(String message, Object... args) {
     }
 
     /**
+     * The method will log messages using the Console Logger. Note only messages that are at least INFO will be printed
+     * The method is needed for publishing messages when the logger initialization data is unknown
+     * Either, before reading the file, or when there was error reading that configuration file
+     * @param requestLevel - The level of the message
+     * @param message - The message to print
+     * @param args - The arguments that are part of the message
+     */
+    public void logConfig(LoggingLevel requestLevel, String message, Object... args) {
+        if (configMode) {
+            if (requestLevel.getValue() >= LoggingLevel.INFO.getValue()) {
+                new ConsoleLoggerOutput().log(createMessage(requestLevel.toString(), message, args));
+            }
```

Comment:
I agree with Yoni that using `configMode` + `logConfig` looks kind of redundant. 
The way I see it is that we could generally simplify this by logging configuration issue through special channel (e.g. the `logConfig` method), and thus we won't need the `configMode` anymore
Also, I'd like to suggest to rename `logConfig` to something more generic (e.g. `adminLog`, `consoleLog`, `directLog`) so we won't be limited for using this channel only for configuration error reporting.